### PR TITLE
Fixes in incident management team builder page

### PR DESCRIPTION
**Pull request by anagperal**
### :pushpin: References

-   **Issue:** Closes #8694tp6ub https://app.clickup.com/t/8694tp6ub

### :memo: Implementation

- [x] Unselected item after delete it
- [x] For the time being, do not allow to delete a parent member in the hierarchy if all children are not also selected, because that member is the one that others have to go to in order to report
- [x] Allow deletion of team roles in bulk except when parents + not all children selected
- [x] Fix search: now user can search by name or by role and tree is flatten when some search term. Also show total selected items when searching if any
- [x] Expand team by default
- [x] Fix bug multiple incident managers
- [x] Change disease outbreak event id in page URL
- [x] Fix infinity loop when you open directly URL without having been on event tracker page before
- [x] Incident Management Team contacts table 
- [x] Do not allow child be in the team hierarchy with itself as parent

### :video_camera: Screenshots/Screen capture

image

Screenshot from 2024-11-07 11-20-25

image
Screenshot from 2024-11-07 14-30-12

[Screencast from 2024-11-07 17-27-51.webm](https://github.com/user-attachments/assets/17902077-c5c1-43d3-b6f5-abe0f506114b)

### CODE REVIEW

File: src/domain/entities/incident-management-team/IncidentManagementTeam.ts [hunk 1]
```diff
@@ -1,10 +1,71 @@
...
+function getAllDescendantsUsernamesByParentUsername(
+    teamMembers: TeamMember[]
+): Map<string, string[]> {
+    const initialMap = teamMembers.reduce((acc, member) => {
+        const entries = (member.teamRoles ?? []).map(role => ({
+            parentUsername: role.reportsToUsername ?? "PARENT_ROOT",
+            username: member.username,
+        }));
+
+        return entries.reduce((mapAcc, { parentUsername, username }) => {
```
Comment (9sneha-n) [inline src/domain/entities/incident-management-team/IncidentManagementTeam.ts:49]: nice!

File: src/webapp/pages/form-page/useForm.ts [hunk 2]
```diff
@@ -65,12 +66,22 @@ export function useForm(formType: FormType, id?: Id): State {
...
+        useState<Maybe<DiseaseOutbreakEvent>>();
     const currentEventTracker = getCurrentEventTracker();
     const { existingEventTrackerTypes } = useExistingEventTrackerTypes();
     useCheckWritePermission(formType);
     const snackbar = useSnackbar();
 
     useEffect(() => {
+        // NOTICE: This if is to avoid infinite loop when the form page is reloaded from browser
+        if (
+            currentEventTrackerState &&
```
Comment (9sneha-n) [inline src/webapp/pages/form-page/useForm.ts:79]: @anagperal With this change, multi step forms stop working. For example, risk and IAP are multi step forms. Clicking on next does not go to the next page. PFA video.  

https://github.com/user-attachments/assets/76afcd18-bebd-4643-903f-6213b507c44e


https://github.com/user-attachments/assets/83817c59-e603-41bc-b2bf-cfbb9c6fc089

**Review by 9sneha-n — CHANGES_REQUESTED** [review]
@anagperal The multi step forms stop working with saving currentEventTracker in local state

File: src/webapp/pages/form-page/useForm.ts (hunk 2 above)
Comment (anagperal) [reply src/webapp/pages/form-page/useForm.ts]: > @anagperal With this change, multi step forms stop working. For example, risk and IAP are multi step forms. Clicking on next does not go to the next page. PFA video.
> 
>  risk-multi-step-not-working.mov 
>  iap-multi-step-not-working.mov

Soled! Check if this is correct https://github.com/EyeSeeTea/zebra-dev/pull/36/commits/368eb3e74c3f99bbf3f08fbec181402bdedea63b

Comment (9sneha-n) [reply src/webapp/pages/form-page/useForm.ts]: @anagperal Thank you! this was a miss on my end , definitely need this change.

**Pull request merged**